### PR TITLE
KAFKA-9396: Use JDK emptyIterator in ConnectHeaders Class

### PR DESCRIPTION
**Pull request by belugabehr**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/api/src/main/java/org/apache/kafka/connect/header/ConnectHeaders.java [hunk 1]
```diff
@@ -265,10 +243,8 @@ public Header lastWithName(String key) {
 
     @Override
     public Iterator<Header> iterator() {
-        if (headers != null) {
-            return headers.iterator();
-        }
-        return EMPTY_ITERATOR;
+        return (headers == null) ? Collections.emptyIterator() :
```
Comment (rondagostino) [inline connect/api/src/main/java/org/apache/kafka/connect/header/ConnectHeaders.java]: nit: parens unnecessary

**Review by rondagostino — COMMENTED** [review]
LGTM as long as tests pass.

**Review by hachikuji — APPROVED** [review]
LGTM. Thanks for the patch!

**Comment by hachikuji** [comment]
retest this please

**Pull request merged**